### PR TITLE
Fix app freeze issue to close #14

### LINKED ISSUE #14: Large numbers are freezing the app
Large numbers calculations, like 9999^9999 cause app to freeze deadly and then crash it, despite "Infinity" answer displays.

https://user-images.githubusercontent.com/110062862/181937798-d3d0d7a6-ad74-4aa3-8d9d-d375ef5660ef.mp4

### PR DESCRIPTION
**Pull request by kabirnayeem99**
Fix app freeze issue with large numbers closes #14

- Introduce Kotlin Coroutine to do all the calculations and string manipulation on the background thread
- Make sure that UI related changes run on the main thread

### CODE REVIEW

File: app/src/main/java/com/darkempire78/opencalculator/MainActivity.kt [hunk 1]
```diff
@@ -136,78 +141,100 @@ class MainActivity : AppCompatActivity() {
...
+                                resultDisplay.setText(result)
+                            }
+                        } else {
+                            withContext(Dispatchers.Main) {
+                                resultDisplay.setText("")
+                            }
+                        }
                     } else {
-                        resultDisplay.setText("")
+                        if (result != calculation) {
```
Comment (danthe1st) [inline app/src/main/java/com/darkempire78/opencalculator/MainActivity.kt:214]: I would personally put the whole if-else blocks in a `withContext` block together.

File: app/src/main/java/com/darkempire78/opencalculator/MainActivity.kt [hunk 2]
```diff
@@ -393,54 +420,59 @@ class MainActivity : AppCompatActivity() {
...
-            val exp = Expression(calculation)
-            var result = exp.calculate().toString()
+                val exp = Expression(calculation)
+                var result = exp.calculate().toString()
+
+                mXparser.consolePrintln("Res: " + exp.expressionString.toString() + " = " + exp.calculate())
 
-            mXparser.consolePrintln("Res: " + exp.expressionString.toString() + " = " + exp.calculate())
+                if (result != "NaN" && result != "Infinity") {
+                    if ((exp.calculate() * 10) % 10 == 0.0) {
```
Comment (danthe1st) [inline app/src/main/java/com/darkempire78/opencalculator/MainActivity.kt:457]: I'd also use a single `withContext` block here for the whole if-else block and move the `calculate`-call up.

Comment (kabirnayeem99) [reply app/src/main/java/com/darkempire78/opencalculator/MainActivity.kt:453]: Thank you for your suggestion, I have made these changes as per your suggestion.

File: app/src/main/java/com/darkempire78/opencalculator/MainActivity.kt (hunk 1 above)
Comment (kabirnayeem99) [reply app/src/main/java/com/darkempire78/opencalculator/MainActivity.kt]: Thank you for your suggestion, I have made these changes as per your suggestion.

**Comment by Darkempire78** [comment]
Thanks

**Pull request merged**